distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Regex/Pattern_2.enso

```diff
@@ -89,6 +91,79 @@ type Pattern_2
     find_all self input =
         self.match_all input . map match_to_group_maybe
 
+    ## ADVANCED
+
+       Replace all occurrences of the pattern described by `self` in the `input`
+       with the specified `replacement`.
+
+       Arguments:
+       - input: The text in which to perform the replacement(s).
+       - replacement: The literal text with which to replace any matches.
+       - only_first: If True, only replace the first match.
+
+       If this method performs no replacements it will return the `input` text
+       unchanged.
+
+       The replacement string can contain references to groups matched by the
+       regex. The following syntaxes are supported:
+           $0: the entire match string
+           $&: the entire match string
+           $nn: group #nn
+           $<foo>: Named group `foo`
+
+       > Example
+         Replace letters in the text "aaa".
+
+             'aaa'.replace 'aa' 'b' == 'ba'
+
+       > Example
+         Replace all occurrences of letters 'l' and 'o' with '#'.
+
+             "Hello World!".replace "[lo]" "#" == "He### W#r#d!"
+
+       > Example
+         Replace the first occurrence of letter 'l' with '#'.
+
+             "Hello World!".replace "l" "#" only_first=True == "He#lo World!"
+
+       > Example
+         Replace texts in quotes with parentheses.
+
+              '"abc" foo "bar" baz'.replace '"(.*?)"' '($1)' matcher=Regex_Matcher == '(abc) foo (bar) baz'
+
+       > Example
+         Replace a literal string with a replacement value.
+
+            pattern = Regex_2.compile "aa"
+            input = "aa ab aa ac ad aa aa ax"
+            match = pattern.replace input "xyz"
+            match . should_equal "xyz ab xyz ac ad xyz xyz ax"
+
+       > Example
+         Replace each word with the same word surrounded by `[]`.
+
+            pattern = Regex_2.compile "([a-z]+)"
+            pattern.replace "foo bar, baz" "[$1]" . should_equal "[foo] [bar], [baz]"
+
+    replace : Text -> Text -> Boolean -> Text
+    replace self input replacement only_first=False =
+        it = Match_Iterator.new self input
+        case it of
+            Match_Iterator_Value.Last filler -> filler.text
+            _ ->
+                replacer = Replacer.new replacement self
+
+                ## Checking if an error to fail fast.
+                if replacer.is_error then replacer else
```

Comment:
Then probably the comment above may not be needed as the `if_not_error` is expressing exactly that - fail fast if there was error, continue otherwise.